test/e2e/fixture/applicationsets/utils/fixture.go

```diff
@@ -210,56 +235,45 @@ func waitForExpectedClusterState() error {
 	}
 
 	// Wait up to 120 seconds for namespace to not exist
-	if err := waitForSuccess(func() error {
-		_, err := fixtureClient.KubeClientset.CoreV1().Namespaces().Get(context.Background(), ApplicationsResourcesNamespace, v1.GetOptions{})
-
-		msg := ""
-
-		if err == nil {
-			msg = fmt.Sprintf("namespace '%s' still exists, after delete", ApplicationsResourcesNamespace)
-		}
-
-		if msg == "" && err != nil && strings.Contains(err.Error(), "not found") {
-			// Success is an error containing 'applicationset-e2e' not found.
-			return nil
-		}
-
-		if msg == "" {
-			msg = err.Error()
+	for _, namespace := range []string{string(ApplicationsResourcesNamespace), string(ArgoCDExternalNamespace), string(ArgoCDExternalNamespace2)} {
```

Comment:
Can we use `fixtureClient.ExternalAppSetClientsets` instead of creating a new list to loop over?